### DISCUSSION TITLE
Float in bootstrap 5 😓😓😓

### DISCUSSION THREAD

**Discussion by bayurzxsmtp** [twbs/bootstrap#35187] [General] [answered]
Whose idea was it to change `float-right and float-left` to `float-start and float-end`. 

Like why???? Do you feel updates are meant to make developers life harder??? Like why please explain. This is why people are talking shit about bootstrap. Saying Tailwind is better. 

I am starting to believe they might have a point. Spent almost 2hrs debugging the most **foolish** of problems.

Thanks to whose idea was it to change it.

**Comment by mitchellholzmann** [comment]
You spend 2 hours debugging something that's literally in the documentation and the migrate documentation as well?

Anyways, the reason for it is that they switched to logical property names instead of directional names. You're completely free to change it back.

**Comment by mdo** [comment, marked as answer]
> Thanks to whose idea was it to change it.

That was me, and you're welcome! 😃 

> Like why???? Do you feel updates are meant to make developers life harder??? Like why please explain.

The answer to your question has been broadly shared. We discussed it in blog posts, release notes, a [migration guide](https://getbootstrap.com/docs/5.1/migration/), [alphas](https://blog.getbootstrap.com/2020/11/11/bootstrap-5-alpha-3/#coming-in-beta-1), [betas](https://blog.getbootstrap.com/2020/12/07/bootstrap-5-beta-1/#rtl), and even tweets. We've literally explained it everywhere we can. We've also explained it in issues and discussions that folks have opened here.

---

Any sarcasm aside, please don't bring the negativity here and assume we're being malicious or difficult in our updates. Major releases for Bootstrap always come with breaking changes, it's been the name of the game for the 10 years it's been around.

**Reply by bayurzxsmtp** [reply]
😂😂😂😂 Your reply was awesome. I was probably having a shitty day then. sorry for being an a**. You guys do too much work to put up with these kinda comments.
It is actually pretty cool. It has grown on me!

Thanks!

**Comment by jipexu** [comment]
Hi
As beginning i was so cold for this changes  "left>start and right>end" and some more breaking change as the data- to data-bs ... and more 
So sure in a CMS context it's give us a lot of work !
But honestly i can say this change are more often logic and follow the langage and media evolution  (eg : start and end in css ) and by chance are well documented in migration guide so it is some work but ... easy work.

So @mdo and all team and contributor thanks for all good work done here, bs5 is a great tool.
